.chloggen/elasticsearchexporter_logs_pipeline_per_event.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: elasticsearchexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add config `logs_dynamic_pipeline` to dynamically set the document pipeline to the value of the attribute `elasticsearch.document_pipeline`, this only applies to logs.
```

Comment:
+1 to only support this for logs right now to avoid the complexities wrt. metrics grouping.

Naming-wise, I was wondering if `elasticsearch.ingest_pipeline` may be a bit clearer. But I don't have a strong opinion on that.